File: src/com/ptby/dynamicreturntypeplugin/config/ConfigStateContainer.java
package com.ptby.dynamicreturntypeplugin.config;

import com.intellij.openapi.project.Project;
import com.ptby.dynamicreturntypeplugin.config.multi.OpenProjectsRefresher;

public class ConfigStateContainer {
    static ConfigState configState = new ConfigState();
    static  OpenProjectsRefresher openProjectsRefresher  = new OpenProjectsRefresher();

    public ConfigStateContainer() {
        openProjectsRefresher = new OpenProjectsRefresher();
    }

    public static ConfigState getConfigState() {
        return configState;
    }



    public static void notifyProjectOpened( Project project ) {
        configState.getOpenProjects().addProject( project );
        configState.getConfigAnalyser().refreshAllConfigs();
    }


    public static void notifyProjectClosed( Project project ) {
        configState.getOpenProjects().removeProject( project );
        configState.getConfigAnalyser().notifyProjectIsClosed( project );
    }
}

File: src/com/ptby/dynamicreturntypeplugin/config/ConfigStateContainer.kt
package com.ptby.dynamicreturntypeplugin.config

import com.intellij.openapi.project.Project

public class ConfigStateContainer {
    class object {
        val configState : ConfigState = ConfigState()


        public fun notifyProjectOpened(project: Project) {
            configState.openProjects.addProject(project)
            configState.configAnalyser.refreshAllConfigs()
        }


        public fun notifyProjectClosed(project: Project) {
            configState.openProjects.removeProject(project)
            configState.configAnalyser.notifyProjectIsClosed(project)
        }
    }
}
